modules/workflow-workflowoperation/src/main/java/org/opencastproject/workflow/handler/workflow/DuplicateEventWorkflowOperationHandler.java

```diff
@@ -0,0 +1,404 @@
+        try (InputStream inputStream = new FileInputStream(sourceFile)) {
+          final URI tmpUri = workspace.put(destination.getIdentifier().toString(), element.getIdentifier(),
+              FilenameUtils.getName(element.getURI().toString()), inputStream);
+          element.setIdentifier(null);
+          element.setURI(tmpUri);
+        }
+
+        // Now we can distribute it to the new media package
+        destination.add(element); // Element has to be added before it can be distributed
+        final Job job = distributionService.distribute(InternalPublicationChannel.CHANNEL_ID, destination,
+            element.getIdentifier());
+        final MediaPackageElement distributedElement =
+            JobUtil.payloadAsMediaPackageElement(serviceRegistry).apply(job);
+        destination.remove(element);
+
+        updateTags(distributedElement, removeTags, addTags, overrideTags);
+
+        PublicationImpl.addElementToPublication(newPublication, distributedElement);
+      } catch (Exception exception) {
+        throw new WorkflowOperationException(exception);
+      }
+    }
+
+    // Using an altered copy of the source publication's URI is a bit hacky,
+    // but it works without knowing the URI pattern...
+    String publicationUri = sourcePublication.getURI().toString();
+    publicationUri = publicationUri.replace(source.getIdentifier().toString(), destination.getIdentifier().toString());
+    publicationUri = publicationUri.replace(sourcePublication.getIdentifier(), newPublicationId);
+    newPublication.setURI(URI.create(publicationUri));
+    destination.add(newPublication);
+  }
+
+  private MediaPackage copyDublinCore(
+      MediaPackage source,
+      MediaPackageElement sourceDublinCore,
+      MediaPackage destination,
+      List<String> removeTags,
+      List<String> addTags,
+      List<String> overrideTags) throws WorkflowOperationException {
+    final DublinCoreCatalog destinationDublinCore = DublinCoreUtil.loadEpisodeDublinCore(workspace, source).get();
+    destinationDublinCore.setIdentifier(null);
+    destinationDublinCore.setURI(sourceDublinCore.getURI());
+    destinationDublinCore.set(DublinCore.PROPERTY_TITLE, destination.getTitle());
+    destinationDublinCore.setFlavor(sourceDublinCore.getFlavor());
+    for (String tag : sourceDublinCore.getTags()) {
+      destinationDublinCore.addTag(tag);
+    }
+    updateTags(destinationDublinCore, removeTags, addTags, overrideTags);
+    try (InputStream inputStream = IOUtils.toInputStream(destinationDublinCore.toXmlString(), "UTF-8")) {
+      final String elementId = UUID.randomUUID().toString();
+      final URI newUrl = workspace.put(destination.getIdentifier().compact(), elementId, "dublincore.xml",
+          inputStream);
+      final MediaPackageElement mpe = destination.add(newUrl, MediaPackageElement.Type.Catalog,
+          MediaPackageElements.EPISODE);
+      mpe.setIdentifier(elementId);
```

Comment:
@krinnewitz In our older version of this code we have found a bug which, as far as I can tell, is still present in the new version: The copied dublincore/episode catalog won't have any tags, will it? 
While you correctly set them for "destinationDublinCore", they are never set to "mpe". Also, you seem to work quite a lot on destinationDublinCore although you only use its "content" (toXmlString()).

Reply:
Yes, thank you. This is now fixed.